roles/ceph-config/tasks/main.yml

```diff
@@ -9,6 +9,27 @@
       group: "ceph"
       mode: "0755"
 
+  - name: count number of osds for non-lvm scenario
+    set_fact:
+      num_osds: "{{ devices | length }}"
+    when:
+      - devices|length > 0
+      - (osd_scenario == 'collocated' or osd_scenario == 'non-collocated')
+
+  - name: count number of osds for lvm scenario
+    set_fact:
+      num_osds: "{{ lvm_volumes | length }}"
+    when:
+      - lvm_volumes|length > 0
+      - osd_scenario == 'lvm'
+
+  - name: count number of osds for lvm-batch scenario
+    set_fact:
+      num_osds: "{{ (ceph-volume lvm batch --report devices | from_json()).osds | length }}"
```

Comment:
adding this here would mean it'd be run on all host, I think it must be run on an OSD.

By the way, you have to do it in two steps :
```
- name: get number of osds for lvm-batch scenario
  command: "{{ docker_exec_cmd }} ceph-volume lvm batch --report devices"
  register: lvm_batch_devices
  when:
    - devices|length > 0 and osd_scenario == 'lvm-batch'
```

```
- name: set_fact num_osds
  set_fact:
    num_osds: "{{ (lvm_batch_devices | from_json).osds | length }}"
  when:
    - devices|length > 0 and osd_scenario == 'lvm-batch'
```